### PR TITLE
#168131422 Users should be able to make multi city request

### PR DESCRIPTION
**Pull request by Dkabiswa**
#### What does this PR do?
- Adds user can perform a multi-city travel request.
#### Description of Task to be completed?
- 
#### How should this be manually tested?
- git clone this repo and checkout this branch
- Run 'npm run db-seed:test` to run new seeds
- run `npm run start:dev` and login valid-user
- Create accommodations in specific locations and use them to perform a request
- Visit `http://localhost:3000/api/v1/requests/multi-city` POST method on postman
```
{
	"from":"Kigali, Rwanda",
	"to": [3, 1],
	"travelDate": ["2019-10-12","2019-10-22"],
	"returnDate": "2019-12-30",
	"reason": "Have been working for 648 days non stop am human not a robot i deserve this cause have earned it",
	"accommodation": ["sheraton", "hotel"]
}
```
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[168131422](https://www.pivotaltracker.com/story/show/168131422)
#### Screenshots (if appropriate)
Screen Shot 2019-09-24 at 09 41 57

#### Questions:
N/A

### CODE REVIEW

**Review by willywax — APPROVED** [review]
Looks good

**Review by Bahatiroben — APPROVED** [review]
changes look good. just resolve the conflicts, and you're good to go.

File: src/routes/api/requests.js [hunk 1]
```diff
@@ -27,11 +28,25 @@ router
...
     requestsValidator.returnTrip,
+    location.validDate,
     location.validLocation,
     location.validAccomodation,
     Requests.trip
   )
   .all(method);
+
+router
+  .route('/multiCity')
```
Comment (Paulvitalis200) [inline src/routes/api/requests.js]: Could you update this to be `multi-city` instead of having it camel case

File: src/swagger.json [hunk 2]
```diff
@@ -870,13 +870,13 @@
         }
       }
     },
-    "/requests/oneWay": {
+    "/requests/returnTrip":{
```
Comment (Paulvitalis200) [inline src/swagger.json]: Make all instances of this endpoint to be `return-trip` and not camel case

File: src/swagger.json [hunk 3]
```diff
@@ -887,10 +887,10 @@
           {
             "in": "body",
             "name": "body",
-            "description": "One way request data entered by the user",
+            "description": "Returb Trip request data entered by the user",
```
Comment (Paulvitalis200) [inline src/swagger.json]: Should be `Return Trip`

File: src/middlewares/validLocation.js [hunk 4]
```diff
@@ -5,6 +5,30 @@ import Response from '../utils/response';
...
+    req.body.travelDate = dates;
+    realDates = dates.map((date) => new Date(date));
+    for (let date = 0; date < realDates.length - 1; date += 1) {
+      if (realDates[date] > realDates[date + 1]) {
+        return Response.customResponse(res, 409, 'travelDates must be in ascending order');
+      }
+    }
+    const returnDate = new Date(req.body.returnDate);
+    if (realDates[realDates.length - 1] > returnDate) {
+      return Response.customResponse(res, 409, 'returnDate must be greater than travel date');
```
Comment (Paulvitalis200) [inline src/middlewares/validLocation.js]: Update this message to read `Return date must be greater than travel date`

File: src/middlewares/validLocation.js [hunk 5]
```diff
@@ -5,6 +5,30 @@ import Response from '../utils/response';
...
+   * @returns {object} The User object.
+   */
+  async validDate(req, res, next) {
+    let realDates = [];
+    const dates = Array.isArray(req.body.travelDate) === true ? req.body.travelDate : [req.body.travelDate];
+    req.body.travelDate = dates;
+    realDates = dates.map((date) => new Date(date));
+    for (let date = 0; date < realDates.length - 1; date += 1) {
+      if (realDates[date] > realDates[date + 1]) {
+        return Response.customResponse(res, 409, 'travelDates must be in ascending order');
```
Comment (Paulvitalis200) [inline src/middlewares/validLocation.js]: `Travel dates must be in ascending order`

**Review by Paulvitalis200 — APPROVED** [review]

**Pull request merged**